pkg/controllers/drift/controller.go

```diff
@@ -0,0 +1,135 @@
+package drift
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/karpenter-core/pkg/apis/provisioning/v1alpha5"
+	operatorcontroller "github.com/aws/karpenter-core/pkg/operator/controller"
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/cloudprovider"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/logging"
+	"net/http"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	"time"
+)
+
+const controllerName = "drift"
+
+type Drift struct {
+	kubeClient    client.Client
+	cloudProvider *cloudprovider.CloudProvider
+}
+
+// NewController constructs a controller instance
+func NewController(kubeClient client.Client, cloudProvider *cloudprovider.CloudProvider) *Drift {
+	return &Drift{
+		kubeClient:    kubeClient,
+		cloudProvider: cloudProvider,
+	}
+}
+
+func (d Drift) Reconcile(ctx context.Context, req controllerruntime.Request) (reconcile.Result, error) {
+	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).Named(controllerName).With("provisioner", req.Name))
+	node := &v1.Node{}
+	if err := d.kubeClient.Get(ctx, req.NamespacedName, node); err != nil {
+		return reconcile.Result{}, nil
+	}
+
+	if _, drifted := node.Annotations[v1alpha5.DriftedAnnotationKey]; drifted {
```

Comment:
You want me to put drifted as a label ? 
And yeah sure I can check the value.

Reply:
And also, not sure if you have noticed but `v1alpha5.DriftedAnnotationKey` belongs to karpenter-core. I hope that's the right thing?

Reply:
I don't see this annotation in Karpenter-core right now, this must be in your fork. I think we'll want to make this a `v1alpha5.DriftedLabelKey` if anything so that we can filter on this.

Reply:
Its there on https://github.com/aws/karpenter-core/pull/57/files

Reply:
@ellistarn How do we manage where the changes belong to karpenter-core also ? The drifted label is part of karpenter-core actually. Do we need to merge that in first ?